bin/hirb.rb

```diff
@@ -15,217 +13,10 @@
-
-# Setup console
-@shell = Shell::Shell.new(@hbase, interactive)
-@shell.debug = @shell_debug
-
-##
-# Toggle shell debugging
-#
-# @return [Boolean] true if debug is turned on after updating the flag
-def debug
-  if @shell_debug
-    @shell_debug = false
-    conf.back_trace_limit = 0
-    log_level = org.apache.log4j.Level::ERROR
-  else
-    @shell_debug = true
-    conf.back_trace_limit = 100
-    log_level = org.apache.log4j.Level::DEBUG
-  end
-  org.apache.log4j.Logger.getLogger('org.apache.zookeeper').setLevel(log_level)
-  org.apache.log4j.Logger.getLogger('org.apache.hadoop.hbase').setLevel(log_level)
-  debug?
-end
-
-##
-# Print whether debug is on or off
-def debug?
-  puts "Debug mode is #{@shell_debug ? 'ON' : 'OFF'}\n\n"
-  nil
-end
-
-
-# For backwards compatibility, this will export all the HBase shell commands, constants, and
-# instance variables (@hbase and @shell) onto Ruby's top-level receiver object known as "main".
-@shell.export_all(self) if top_level_definitions
-
-# If script2run, try running it.  If we're in interactive mode, will go on to run the shell unless
-# script calls 'exit' or 'exit 0' or 'exit errcode'.
-require 'shell/hbase_loader'
-if script2run
-  ::Shell::Shell.exception_handler(!$fullBackTrace) { @shell.eval_io(Hbase::Loader.file_for_load(script2run), filename = script2run) }
-end
-
-# If we are not running interactively, evaluate standard input
-::Shell::Shell.exception_handler(!$fullBackTrace) { @shell.eval_io(STDIN) } unless interactive
-
-if interactive
-  # Output a banner message that tells users where to go for help
-  @shell.print_banner
-
-  require 'irb'
-  require 'irb/ext/change-ws'
-  require 'irb/hirb'
-
-  module IRB
-    # Override of the default IRB.start
-    def self.start(ap_path = nil)
-      $0 = File.basename(ap_path, '.rb') if ap_path
-
-      IRB.setup(ap_path)
-      @CONF[:IRB_NAME] = 'hbase'
-      @CONF[:AP_NAME] = 'hbase'
-      @CONF[:BACK_TRACE_LIMIT] = 0 unless $fullBackTrace
-
-      hirb = if @CONF[:SCRIPT]
-               HIRB.new(nil, @CONF[:SCRIPT])
-             else
-               HIRB.new
-             end
-
-      shl = TOPLEVEL_BINDING.receiver.instance_variable_get :'@shell'
-      hirb.context.change_workspace shl.get_workspace
-
-      @CONF[:IRB_RC].call(hirb.context) if @CONF[:IRB_RC]
-      # Storing our current HBase IRB Context as the main context is imperative for several reasons,
-      # including auto-completion.
-      @CONF[:MAIN_CONTEXT] = hirb.context
-
-      catch(:IRB_EXIT) do
-        hirb.eval_input
-      end
-    end
-  end
-
-  IRB.start
-end
+puts <<EOF
+  This file has been superceded by packaging our ruby files into a jar
+  and using jruby's bootstrapping to invoke them. If you need to
+  source this file fo some reason it is now named 'jar-bootstrap.rb' and is
+  located in the root of the file hbase-shell.jar and in the source tree at
+  'hbase-shell/src/main/ruby'.
+EOF
```

Comment:
The command-line options still work?

Reply:
Nice being able to remove this.

Reply:
No. Just asking. I think I figured it myself later after reading more... that this stuff would still work.